sklearn/impute/_base.py

```diff
@@ -52,9 +52,7 @@ def _most_frequent(array, extra_value, n_repeat):
                 if count == most_frequent_count
             )
         else:
-            mode = stats.mode(array)
-            most_frequent_value = mode[0][0]
-            most_frequent_count = mode[1][0]
+            most_frequent_value, most_frequent_count = stats.mode(array)
```

Comment:
I think we need to work around the `FutureWarning` that `stats.mode` will raise when `keepdims` is not set?

https://github.com/scipy/scipy/blob/7ed8128f86dadf9a9a912f0699576dbeadb02638/scipy/stats/_stats_py.py#L644-L653